### ISSUE TITLE
Feature request for modifying ending tags automatically

### ISSUE THREAD

**Issue by ShashankaNataraj** [Microsoft/vscode#11153]
- VSCode Version:1.4.0
- OS Version:0.11.6

Steps to Reproduce:

1. Have this in your html code: `<h5>Hello, World</h5>`
2. Now try to modify `h5` to `H6`
3. The ending tag doesn't automatically change to `H6` like it does in webstorm

**Comment by bastienJS** [comment]
feature request? thats a bug...

**Comment by kenisteward** [comment]
How can I tell if this is currently being worked on / If i can contribute?  I wanted to look into this.

**Comment by aeschli** [comment]
It's not being worked on. You're help is highly welcome, but the functionality yur request is quite tricky to implement in our architecture.
The knowledge of matching tags is only in the language server, implemented [here](https://github.com/Microsoft/vscode-html-languageservice), while the changes to open documents would have to be made in the html extension or even in the renderer.

**Comment by kenisteward** [comment]
@aeschli Thanks for giving me a starting point.  I'll report back here if I am successful.

**Comment by ShashankaNataraj** [comment]
@aeschli Understood, but this is something every paid editor out there implements. VSCode is better than many of them but this is one feature I sorely miss!

**Comment by bermanboris** [comment]
+1

**Comment by kenisteward** [comment]
@aechli the extension ( even though helpful ) seems to still have odd quirks ( typing too fast bug / performance issues sometimes ). 

Even though this is closed would PR'S still be accepted for native support? I wouldn't mind trying with that extension as a a base as I have more time over the holidays.

**Comment by aeschli** [comment]
@kenisteward Yes sure, that would be cool! And I'm sure also @formulahendry welcomes help with the auto-rename extension. Please don't forget to file the issues you found there.

**Comment by lzl124631x** [comment]
+1. Still no built-in support yet?
